### ISSUE TITLE
Internal Server Error After Upgrading To Kubernetes 1.23

### ISSUE THREAD

**Issue by jimitkr** [kubernetes-sigs/prometheus-adapter#563]
I recently upgraded my Kubernetes version to 1.23 and have been getting the following error on the custom metrics api since then:
```

$ kubectl get --raw "/apis/custom.metrics.k8s.io/v1beta1"
Error from server (InternalError): an error on the server ("Internal Server Error: \"/apis/custom.metrics.k8s.io/v1beta1\": the server could not find the requested resource") has prevented the request from succeeding
``` 

Kubernetes api-server has the following error:
```
E0209 05:15:17.882999      11 controller.go:116] loading OpenAPI spec for "v1beta1.custom.metrics.k8s.io" failed with: failed to retrieve openAPI spec, http error: ResponseCode: 500, Body: Internal Server Error: "/openapi/v2": the server could not find the requested resource
, Header: map[Content-Length:[87] Content-Type:[text/plain; charset=utf-8] Date:[Thu, 09 Feb 2023 05:15:17 GMT] X-Content-Type-Options:[nosniff]]
```

My API service is configured as follows:

```
apiVersion: apiregistration.k8s.io/v1
kind: APIService
metadata:
  name: v1beta1.custom.metrics.k8s.io
spec:
  service:
    name: prometheus-adapter
    namespace: monitoring
  group: custom.metrics.k8s.io
  version: v1beta1
  insecureSkipTLSVerify: true
  groupPriorityMinimum: 100
  versionPriority: 100
```

What am i doing wrong?

**Comment by jimitkr** [comment]
@olivierlemasle I see a new error today which might provide more insight. I've added this error to the main body of the issue as well.

```
$ kubectl get --raw "/apis/custom.metrics.k8s.io/v1beta1"
Error from server (InternalError): an error on the server ("Internal Server Error: \"/apis/custom.metrics.k8s.io/v1beta1\": subjectaccessreviews.authorization.k8s.io is forbidden: User \"system:serviceaccount:monitoring:prometheus-adapter\" cannot create resource \"subjectaccessreviews\" in API group \"authorization.k8s.io\" at the cluster scope") has prevented the request from succeeding
```

**Comment by jimitkr** [comment]
@olivierlemasle Sorry to bother. Did you have any luck replicating the problem?

**Comment by jimitkr** [comment]
@olivierlemasle My issue was fixed after upgrading prometheus adapter to v0.10.0
